components/SwipeRow.js

```diff
@@ -62,6 +65,42 @@ class SwipeRow extends Component {
 				this.previousTrackedDirection = direction;
 			});
 		}
+
+		if (this.props.forceCloseToRightThreshold && this.props.forceCloseToRightThreshold > 0) {
+			this._translateX.addListener(({ value }) => {
+				if(!this.state.isForceClosing && (Dimensions.get('window').width + value) < this.props.forceCloseToRightThreshold) {
+					this.setState({
+						isForceClosing: true,
+						forceClosingDirection: "right"
+					});
+					/* Must be different thread to prevent the infinite loop, but to be able to see the new state */
+					setTimeout(() => {
+						this.forceCloseRow();
+						if(this.props.onForceCloseToRight) {
+							this.props.onForceCloseToRight();
+						}
+					}, 0);
```

Comment:
Rather than adding `isForceClosing` and `forceClosingDirection` to `state` can they just be instance variables like `this. isForceClosing = true`. That way you don't have to worry about waiting until state is updated and can remove the `setTimeout`  ?

Reply:
@jemise111 Oh! interesting! I'll try it out!!

Reply:
@jemise111 Done! https://github.com/jemise111/react-native-swipe-list-view/pull/299/commits/4ce259301df44a92dac641919febfa10c9e8f407